src/typeahead/typeahead.ts

```diff
@@ -312,7 +313,7 @@ export class NgbTypeahead implements ControlValueAccessor,
   }
 
   private _showHint() {
-    if (this.showHint) {
+    if (this.showHint && this._userInput != null) {
```

Comment:
What is the exact condition under which `this._userInput` gets `null`? Maybe we should rather assume that user 's input is always a string, instead of checking this in several places?